File: code_samples/go/networking_tcp_udp.go
package main

import (
    "fmt"
)

// Networking: Tcp Udp
// AI/ML Training Sample

type TcpUdp struct {
    Data string
}

func NewTcpUdp() *TcpUdp {
    return &TcpUdp{
        Data: "",
    }
}

func (s *TcpUdp) Process(input string) {
    s.Data = input
}

func (s *TcpUdp) Validate() bool {
    return len(s.Data) > 0
}

func (s *TcpUdp) GetData() string {
    return s.Data
}

func main() {
    instance := NewTcpUdp()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/lua/networking_tcp_udp.lua
-- Networking: Tcp Udp
-- AI/ML Training Sample

TcpUdp = {}
TcpUdp.__ index = TcpUdp

function TcpUdp:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function TcpUdp:process(input)
    self.data = input
end

function TcpUdp:getData()
    return self.data
end

function TcpUdp:validate()
    return #self.data > 0
end

-- Example usage
local instance = TcpUdp:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
